packages/web/utils/stringHelpers.ts

```diff
@@ -7,3 +11,57 @@ export const hashCode = (str: string): string => {
   }
   return hash.toString();
 };
+
+/**
+ * sanitize string of HTML & parse it into jsx elements
+ * */
+export const safelyParseContent = (content: string) => {
+  const clean = DOMPurify.sanitize(content);
+  const options = {
+    replace: (domNode: any) => {
+      if (domNode.attribs?.href) {
+        domNode.attribs.target = '_blank';
+        domNode.attribs.title = `Opens new tab to ${domNode.attribs.href}`;
+        domNode.attribs.class = 'external-link';
+      }
+    },
+  };
+
+  const parsed = parse(clean, options);
+  return parsed;
+};
+
+/**
+ * Takes a string of HTML elements, sanitizes it,
+ * applies classNames to the elements & returns parsed HTML as
+ * `JSX.Element | JSX.Element[] | string` */
+export const safelyParseTextForTyping = (content: string) => {
+  const clean = DOMPurify.sanitize(content);
+  const options = {
+    replace: (domNode: any) => {
+      if (domNode.type === 'tag' && domNode.children.length > 0) {
+        if (domNode.parent === null) {
+          domNode.attribs.class = 'typing-text';
+        }
+        if (domNode.attribs?.href) {
+          domNode.attribs.target = '_blank';
+          domNode.attribs.title = `Opens new tab to ${domNode.attribs.href}`;
+          domNode.attribs.class = 'external-link';
+        }
+      }
+    },
+  };
+
+  const parsed = parse(clean, options);
+  return parsed;
+};
+
+export const shortenAddress = (address: string, sliceSize?: number): string => {
+  const sliceNumber = sliceSize ?? 4;
+  const start: string = address.toLowerCase().slice(0, sliceNumber);
+  const end: string = address
+    .toLowerCase()
+    .slice(Math.max(0, address.length - sliceNumber));
```

Comment:
Why do you lower case it? I'd not.

```suggestion
  const start: string = address.slice(0, sliceNumber);
  const end: string = address
    .slice(Math.max(0, address.length - sliceNumber));
```